lib/base/communication.js

```diff
@@ -28,29 +29,46 @@ var sendCommandToSocket = function (options, socket, callback) {
     });
 
     socket.on('end', function () {
+        var messages = [];
+        var buffer = Buffer.concat(buffers);
+        debug('socket.on end, Total length', buffer.length);
+        while (buffer.length >= headerLength) {
+            var header = {};
+            for (var i = 0; i < headerProps.length; i++) {
+                var prop = headerProps[i];
+                header[prop] = ntohl(buffer, i * 4);
+            }
+            debug('Extracted header', header);
+            if (header.ret < 0) {
+                callbackOnce({
+                    msg: 'Communication Error. Received ' + header.ret,
+                    header: header,
+                    options: options
+                });
+            }
+            if (header.payload > 0) {
+                var payload = buffer.slice(24,24+header.payload).toString('utf8');
```

Comment:
I have changed it in my running system, I can't see how it should cause any problems but want it to run overnight to make sure the condition that caused the problem will occur to make sure it is ok.